integration/local-gadget/list_containers_test.go

```diff
@@ -178,3 +182,60 @@ func TestWatchContainers(t *testing.T) {
 
 	RunCommands(commands, t)
 }
+
+func TestWatchDeletedContainers(t *testing.T) {
+	t.Parallel()
+	cn := "test-deleted-container"
+	ns := GenerateTestNamespaceName(cn)
+
+	// TODO: Handle it once we support getting K8s container name for docker
+	// Issue: https://github.com/inspektor-gadget/inspektor-gadget/issues/737
+	if *containerRuntime == ContainerRuntimeDocker {
+		t.Skip("Skip TestWatchContainers on docker since we don't propagate the Kubernetes pod container name")
+	}
+
+	watchContainersCmd := &Command{
+		Name:         "RunWatchContainers",
+		Cmd:          fmt.Sprintf("local-gadget list-containers -o json --runtimes=%s --containername=%s --watch", *containerRuntime, cn),
+		StartAndStop: true,
+		ExpectedOutputFn: func(output string) error {
+			expectedEvent := &containercollection.PubSubEvent{
+				Type: containercollection.EventTypeRemoveContainer,
+				Container: &containercollection.Container{
+					Name:      cn,
+					Podname:   cn,
+					Runtime:   *containerRuntime,
+					Namespace: ns,
+				},
+			}
+
+			normalize := func(e *containercollection.PubSubEvent) {
+				e.Container.ID = ""
+				e.Container.Pid = 0
+				e.Container.OciConfig = nil
+				e.Container.Bundle = ""
+				e.Container.Mntns = 0
+				e.Container.Netns = 0
+				e.Container.CgroupPath = ""
+				e.Container.CgroupID = 0
+				e.Container.CgroupV1 = ""
+				e.Container.CgroupV2 = ""
+				e.Container.Labels = nil
+				e.Container.PodUID = ""
+				e.Timestamp = ""
+			}
+
+			return ExpectEntriesToMatch(output, normalize, expectedEvent)
+		},
+	}
+
+	commands := []*Command{
+		CreateTestNamespaceCommand(ns),
+		PodCommand(cn, "busybox", ns, `["sleep", "5"]`, ""),
+		WaitUntilPodReadyCommand(ns, cn),
+		watchContainersCmd,
+		DeleteTestNamespaceCommand(ns),
+	}
```

Comment:
Isn't it a little bit tricky? I mean, it relies on the fact that the `watchContainersCmd` will last 10 seconds to capture the deletion. What about:

```suggestion
	commands := []*Command{
		CreateTestNamespaceCommand(ns),
		PodCommand(cn, "busybox", ns, `["sleep", "inf"]`, ""),
		WaitUntilPodReadyCommand(ns, cn),
		watchContainersCmd,
		SleepForSecondsCommand(2), // wait to ensure local-gadget has started
		{
			Name:         "DeletePod",
			Cmd:          fmt.Sprintf("kubectl delete pod %s -n %s", cn, ns),
		},
		DeleteTestNamespaceCommand(ns),
	}
```

Reply:
Agree, this is much more clear. Thanks for the suggestion.